src/sources/ldap.rs

```diff
@@ -90,9 +90,9 @@ impl LdapSource {
 			.any(|flag| status_as_int & flag != 0);
 
 		let ldap_user_id = match read_search_entry(&entry, &self.ldap_config.attributes.user_id)? {
-			// TODO(tlater): Use an encoding that preserves alphabetic order
-			StringOrBytes::Bytes(byte_id) => BASE64_STANDARD.encode(byte_id),
-			StringOrBytes::String(string_id) => BASE64_STANDARD.encode(string_id),
+			// Use hex encoding instead of base64 for consistent alphabetical order
+			StringOrBytes::Bytes(byte_id) => hex::encode(byte_id),
+			StringOrBytes::String(string_id) => hex::encode(string_id.to_lowercase().as_bytes()),
```

Comment:
Are (open)ldap IDs not case sensitive?

Reply:
Not sure about this. Let's discuss.

Reply:
Well, if they are case sensitive this will cause collisions, while if it they are case sensitive there's a chance that they might change the casing randomly.

I assume the latter is probably not the case, and we should just remove the `to_lowercase()`. Is there a particular reason we do that?

Reply:
Since there is not a particular reason, I agree with you - it's logical to remove the lowercase conversion.